src/ui/ControlDescription.tsx

```diff
@@ -0,0 +1,170 @@
+  VStack,
+  Text,
+  useToast
+} from "@chakra-ui/react";
+import { Endpoint } from "../utils/types"
+import Context from "./context";
+//import AutoSizer from "react-virtualized-auto-sizer";
+import classes from "./controlDynamic.module.css";
+
+interface EndpointsByHost {
+  [host: string]: Endpoint[];
+}
+
+
+const ControlDescription = () => {
+  const context = useContext(Context);
+  //const [search, setSearch] = useState("");
+  const [apiKey, setApiKey] = useState("");
+  const toast = useToast();
+
+  // Load the API key from local storage when the component mounts
+  useEffect(() => {
+    const storedApiKey = localStorage.getItem('OPENAI_API_KEY');
+    if (storedApiKey) {
+      setApiKey(storedApiKey);
+    }
+  }, []);
+
+  // Handler to update local storage when the API key changes
+  const handleApiKeyChange = (newApiKey: string) => {
+    localStorage.setItem('OPENAI_API_KEY', newApiKey);
+    setApiKey(newApiKey);
+  };
+
+  // This will be an array of strings representing the values of the selected checkboxes
+  const selectedEndpoints: string[] = Array.from(context.selectedEndpoints);
+
+  const onCheckboxChange = (value: string) => {
+    // Create a new Set from the existing one
+    const newSelectedEndpoints = new Set(selectedEndpoints);
+    if (newSelectedEndpoints.has(value)) {
+      newSelectedEndpoints.delete(value);
+    } else {
+      newSelectedEndpoints.add(value);
+    }
+    context.setSelectedEndpoints(newSelectedEndpoints);
+  };
+
+  // Use `EndpointsByHost` to type the accumulator
+  const endpointsByHost = context.endpoints.reduce((acc: EndpointsByHost, endpoint: Endpoint) => {
+      const { host } = endpoint;
+      if (!acc[host]) {
+        acc[host] = [];
+      }
+      acc[host].push(endpoint);
+      return acc;
+    }, {} as EndpointsByHost);
+
+
+  const handleDescribeEndpoints = () => {
+    if (!apiKey) {
+      toast({
+        title: "API Key Required",
+        description: "Please enter a valid OpenAI API key before proceeding.",
+        status: "error",
+        duration: 5000,
+        isClosable: true,
+      });
+      return;
+    }
+    context.describeSelectedEndpoints(context.selectedEndpoints);
+  };
+    
+  return (
+    <>
+      <VStack spacing={4}>
+
+        <div className={classes.wrapper}>
+          
+          <Heading as="h2" size="md" marginBottom="1em" marginTop="1em">
+            Select Endpoints for AI Description
+          </Heading>
+          <Heading as="h2" size="sm" marginBottom="1em" marginTop="1em">
+            Your OpenAI key is only ever stored locally
+          </Heading>
+          <Input
+            placeholder="Enter your OpenAI API Key. This is only ever stored locally."
+            value={apiKey}
+            onChange={(e) => handleApiKeyChange(e.target.value)}
+            className={classes.apiKeyInput}
+            marginBottom="1em"
+          />
+
+          {context.endpoints.length !== 0 && context.selectedEndpoints.size === 0 && (
```

Comment:
I would leave this where it is instead of hiding when all are selected - the jumping is unpleasant

https://github.com/Superflows-AI/openapi-llm/assets/33871096/c4ad2343-4f92-4456-ab7d-239b9adcbc4a